### PR TITLE
Adding Users and Profiles test data and unit test with pytest

### LINKED ISSUE #55: Adding Profile envelope function and tests
Adding Profile envelope function  and unit tests with pytest
fixes #74

### PR DESCRIPTION
**Pull request by Davidco94**
1) Adding Users and Profiles test data to be created when using migrate.
2) Creating superuser account on migrate commend.
3) Adding Users and Profiles unit test with pytest.

fixes #55

### CODE REVIEW

**Review by shirazyomtov — APPROVED** [review]
LGTM

File: users/tests.py [hunk 1]
```diff
@@ -1,3 +1,171 @@
...
+FIRSTNAME = "Test"
+LASTNAME = "User"
+PASSWORD = "testpass"
+EMAIL = "testuser@gmail.com"
+PHONE_NUMBER = "0541234567"
+LONG_PROFESSION = " test test test test test test test test test test test test test test test test" \
+                 " test test test test test test test test test" \
+                 " test test test test test test test test test" \
+                 " test test test test "
+LONG_BIO = """ test test test test test test test test test test test test test test test test
```
Comment (Omeramsc) [inline users/tests.py:23]: what are we trying to achieve using this long text?

File: users/tests.py [hunk 2]
```diff
@@ -1,3 +1,171 @@
...
+    return new_second_user
+
+
+@pytest.mark.django_db()
+class TestUserModel():
+    # Check if the user is saved in the database
+    def test_persist_user(self, new_user):
+        new_user.save()
+        assert new_user in User.objects.all()
+        assert new_user.profile in Profile.objects.all()
```
Comment (Omeramsc) [inline users/tests.py:65]: no need to test dJango builtins, the dJango folks probably already tested this method to be working.

File: users/tests.py [hunk 3]
```diff
@@ -1,3 +1,171 @@
...
+
+    # Check that the user values are the same as the user inputs.
+    def test_new_user_input_validation_with_db(self, new_user):
```
Comment (Omeramsc) [inline users/tests.py:68]: same

File: users/tests.py [hunk 4]
```diff
@@ -1,3 +1,171 @@
...
+                        first_name=FIRSTNAME,
+                        last_name=LASTNAME,
+                        password=PASSWORD,
+                        email=EMAIL)
+            user.save()
+            user.profile.phone_number = PHONE_NUMBER
+            user.profile.clean_fields()
+
+    # Test for invalid city
+    def test_invalid_city_length(self, new_user):
```
Comment (Omeramsc) [inline users/tests.py:146]: Those tests are OK though, while they kind-of testing Django builtins, they can be used to assure a change to the city_length, or registering a test-entity with the wrong length, will be handled.

**Review by Omeramsc — CHANGES_REQUESTED** [review]
I left a comment on slack instead of repeating the same thing about builtins testing, please take a look.

File: users/tests.py (hunk 1 above)
Comment (Davidco94) [reply users/tests.py:23]: bio max length is 500 and max profession length is 100 so this long text is to exceed the bio max length

File: users/tests.py (hunk 2 above)
Comment (Davidco94) [reply users/tests.py]: I deleted this

File: users/tests.py (hunk 3 above)
Comment (Davidco94) [reply users/tests.py]: I deleted this

File: users/tests.py (hunk 4 above)
Comment (Davidco94) [reply users/tests.py:119]: this is the purpose of the function here to check invalid input

**Review by shirazyomtov — APPROVED** [review]
LGTM

File: users/tests.py [hunk 5]
```diff
@@ -1,3 +1,144 @@
...
+
+    # In this test we delete only the profile part, and check if the profile was deleted and not the whole user
+    def test_delete_only_profile_from_db(self, new_user):
+        new_user.save()
+        new_user.profile.delete()
+        assert new_user in User.objects.all()
+        assert new_user.profile not in Profile.objects.all()
+
+    # Check invalid email
+    def test_validate_email_addr(self):
```
Comment (natifridman) [inline users/tests.py:70]: This is already being tested by the built-in django user, no need for this

File: users/tests.py [hunk 6]
```diff
@@ -1,3 +1,144 @@
...
+            user = User(
+                        username=USERNAME,
+                        first_name=FIRSTNAME,
+                        last_name=LASTNAME,
+                        password=PASSWORD,
+                        email="check")
+            user.full_clean()
+
+    # Test for unique username
+    def test_unique_username(self):
```
Comment (natifridman) [inline users/tests.py:81]: same

File: users/tests.py [hunk 7]
```diff
@@ -1,3 +1,144 @@
...
+def persist_second_user(new_second_user):
+    new_second_user.save()
+    return new_second_user
+
+
+@pytest.mark.django_db()
+class TestUserModel():
+
+    # In this test we delete only the profile part, and check if the profile was deleted and not the whole user
+    def test_delete_only_profile_from_db(self, new_user):
```
Comment (natifridman) [inline users/tests.py:63]: What if you delete the user? will the profile be deleted?

**Review by Davidco94 — COMMENTED** [review]
I'm working on another function that I'm doing some tests for so I'll upload the changes together when everything works

File: users/tests.py (hunk 1 above)
Comment (Omeramsc) [reply users/tests.py:23]: I would rather have here something like `"test_exceeding_max_length"*100` or something like this

**Comment by natifridman** [comment]
Any update on this? it's waiting for a long time

**Comment by Davidco94** [comment]
I deleted all these tests, I will upload a new PR

**Pull request closed without merging**